R/merge_sce_list.R

```diff
@@ -0,0 +1,138 @@
+#' Merge a list of SCEs as preparation for formal integration
+#'
+#' This function takes a named (ideally by a form of library ID) list of
+#'  SingleCellExperiment (SCE) objects and merges them into one SCE object.
+#'  All names in the colData slot must match across provided SCE objects.
+#'  The resulting colData slot will contain information in the given batch column
+#'  to differentiate originating SCE objects. Only features (genes) that are present
+#'  in all provided SCE objects will be retained. To preserve original feature-level
+#'  information, all final rowData slot column names will be appended with the
+#'  given SCE's name, as `<column_name>-<sce_name>`.
+#'
+#' @param sce_list A named list of SingleCellExperiment objects
+#' @param batch_column A character value giving the resulting colData column name
+#'  to differentiate originating SingleCellExperiment objects Default value is `batch`.
+#' @param retain_coldata_columns_main A vector of colData columns which should be retained
+#'  in the main experiment of the final merged SCE object.
+#' @param retain_coldata_columns_altexps A vector of colData columns which should be retained
+#'  in the altExps experiments of the final merged SCE object.
+#' @param retain_altexps A logical indicating whether any present altExp slots should
+#'   be retained in the merged SCE. Default is `TRUE`.
+#' @param preserve_rowdata_columns A vector of column names that appear in originating
+#'  SingleCellExperiment objects' rowData slots which should not be renamed with
+#'  the given SingleCellExperiment object name. These are generally columns which are
+#'  not specific to the given library's preparation or statistics. For example,
+#'  such an array might contain items like "Gene", "ensembl_ids", "gene_symbol", etc.
+#'
+#' @return A SingleCellExperiment object containing all SingleCellExperiment objects
+#'   present in the inputted list
+#' @export
+#'
+#' @import SingleCellExperiment
+merge_sce_list <- function(sce_list = list(),
+                           batch_column = "batch",
```

Comment:
I'm torn on this... if you remember our lengthy discussion on what a batch is, I wonder if this is confusing or if we should this opportunity to switch to `library_id` instead? Technically we are using the `library_id` and putting it into a column named `batch`, but everywhere else in `scpca-nf` and in `scpcaTools` we refer to it with `library_id` which is why I might slightly favor changing to that over batch...

Reply:
Interesting.. In my mind `batch` is a much more general term to use here than `library_id` but consistency within the ScPCA universe is important! 
My reading of your comment is to change the default value to "library_id" but keep the argument `batch_column` name? Yes/no?

Reply:
Yes, I think that's what I'm getting to. I just want to make sure we have an understanding that "batch" here is in reality just the library ID, or what we have been using as library ID throughout scpca. So I like that solution of `batch_column` with the default of `library_id`.